backends/s3/s3.go

```diff
@@ -45,6 +45,9 @@ type Options struct {
 	AccessKey string `yaml:"access_key"`
 	SecretKey string `yaml:"secret_key"`
 
+	// Allow using custom credentials provider.
+	Provider K8sSecretProvider `yaml:"kubernetes_secrets,omitempty"`
```

Comment:
It would be cleaner to use a top-level `access_key_file` and `secret_key_file` attribute and create the custom provider from these when needed. This would decouple the configuration from the implementation details and avoid an error-prove extra indentation level in the YAML config.

There is also a naming mismatch:

- variable: Provider (very non-descriptive)
- type: K8sSecretProvider
- json: kubernetes_secret